packages/shared/src/components/auth/AuthDefault.tsx

```diff
@@ -51,6 +53,11 @@ const AuthDefault = ({
     return onSignup(input.value.trim());
   };
 
+  const onSocialClick = (provider: string) => {
+    storage.setItem(SIGNIN_METHOD_KEY, provider);
+    onProviderClick(provider);
+  };
```

Comment:
We also need this on the register calls, perhaps we can do it inside the final onRegister and onLogin?

I would also suggest if we can convert to cookie?
That way it will sync across apps

Reply:
Thinking about it, can't we have Kratos set a cookie?

Reply:
But for login, I think there should not be any difference in the UI since only the placement of the providers was arranged. What do you think?

Reply:
> Thinking about it, can't we have Kratos set a cookie?

I don't want to involve Kratos now as much as possible, to be honest 🤣

Reply:
Not sure I understand the login query.
To me it should be for register and login -> set cookie -> next login we'll show the different login UI?

Reply:
Yes, that's correct. Though I thought it would only be when the user logged in with the social provider. Let get back to the specs again.

Reply:
For this one, I'm not sure which social provider we would display if the user logged out from password method.

![image](https://user-images.githubusercontent.com/13744167/185538956-7a348b1b-5cc9-4fe1-9885-d584167afe4b.png)

Reply:
Let's raise it for product/design please 🙏

Reply:
After discussing with Tomer, we can go with the default for if the last used method was login.

https://dailydotdev.slack.com/archives/C03KATZEKDW/p1660933534458229?thread_ts=1660900677.896609&cid=C03KATZEKDW

Reply:
I think we had a misunderstanding here.
My point was more around the social flow, both register and signin with social should remember the state.
Not sure if that's the case now.

And still think we should aim for a cookie to do cross system?

Reply:
Ah, yes. This should remember the state. The current flow is when the user clicked any provider it would already register what they chose even before finishing the login/registration.